src/api/procedures/utils.ts

```diff
@@ -710,3 +712,51 @@ export async function addManualFees(
 
   return fees.reduce((prev, { fees: nextFees }) => prev.plus(nextFees), currentFee);
 }
+
+/**
+ * Checks that Metadata exists and that it's value is not locked
+ * @hidden
+ */
+export async function assertMetadataValueIsModifiable(metadataEntry: MetadataEntry): Promise<void> {
```

Comment:
Could this be a method on the MetadataEntry it self?

I think it could be exposed as `isModifiable` and then for the procedure throw the error in the procedure if false..

That way on a UI the same logic can be used to disable a button and for stopping to procedure